### DISCUSSION TITLE
Unable to upload image for model prediction

### DISCUSSION THREAD

**Discussion by sparshgarg23** [fastapi/fastapi#8469] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import os
import cv2 as cv
from fastapi import FastAPI, UploadFile, File
import uvicorn
from fastapi.encoders import jsonable_encoder

app=FastAPI()
@app.post("/create_file/")
async def image(image:UploadFile=File(...)):
    print(image.file)

    try:
        os.mkdir("images")
        print(os.getcwd())
    except Exception as e:
        print(e)
    file_name=os.getcwd()+"/images/"+image.filename.replace(" ","-")

    with open(file_name,'wb+') as f:
        f.write(image.file.read())
        f.close()
    
    img = cv.imread(file_name)
    gray=cv.cvtColor(img,cv.COLOR_BGR2GRAY)
    file_name1=os.getcwd()+"/images/"+"grayscale"+image.filename.replace(" ","-")
    cv.imwrite(file_name1,gray)
    
    #cv.imshow('Image',img)
    #cv.imshow('Gray',gray)
    file=jsonable_encoder({"imagePath":file_name1})
    #new_image=await add_image(file)
    return file

if __name__=="__main__":
    uvicorn.run(app,host="127.0.0.1",port=8000)
```

### Description

Open the browser and call /create_files/
I am getting method not allowed 

### Operating System

Windows

### FastAPI Version

0.75.1

### Python Version

3.9.7

### Additional Context

![fastapi_upload_error](https://user-images.githubusercontent.com/34626942/165092632-4f722357-c173-4a3d-8bad-d79875a7f455.png)
)

**Comment by jgould22** [comment]
You have defined a HTTP POST, but when you open it in your browser like that it is sending a GET request.

**Comment by sparshgarg23** [comment]
I changed method to get and then went to http://127.0.0.1:8000/docs after uploading image I received the following error
TypeError: Failed to execute 'fetch' on 'Window': Request with GET/HEAD method cannot have body.
If I go to the link http://127.0.0.1:8000 then i get the error
{"detail":[{"loc":["body","image"],"msg":"field required","type":"value_error.missing"}]}

**Comment by sparshgarg23** [comment]
@jgould22  I also tried running the following snippet
```
from fastapi import FastAPI, UploadFile, File, Form
from PIL import Image
from io import BytesIO
import uvicorn
import numpy as np
import cv2 as cv2

app = FastAPI()

def load_image_into_numpy_array(data):
    npimg=np.frombuffer(data,np.uint8)
    frame=cv2.imdecode(npimg,cv2.IMREAD_COLOR)
    return cv2.cvtColor(frame,cv2.COLOR_BGR2RGB)
    	
    #return np.array(Image.open(BytesIO(data)))


@app.post("/")
async def read_root(file: UploadFile = File(...)):
    image = load_image_into_numpy_array(await file.read())
    print(image)	
    return {"Hello": "World"}
if __name__=="__main__":
    uvicorn.run(app,host="127.0.0.1",port=8000)
```
I am facing the same issue with this code,when I pass image via curl,it's being read succesfully and I am able to both print the image and see Hello world.The same output is observed when I go to http://127.0.0.1:8000/docs
But when I go to http://127.0.0.1:8000 i end up getting method not allowed

**Comment by jgould22** [comment]
When you place `http://127.0.0.1:8000/` in the navigation bar and hit enter the browsers default behavior is to send a GET request to address you entered. This is expected behavior. 

Since you have not defined a handler for GET FastAPI tells you that the method is not allowed.

What is the behaviour you expect?

**Comment by sparshgarg23** [comment]
My expected behaviour is as follows
When i go to 127.9.9.1:8000/ I should be able to upload an image,and then in the output console its prediction should be displayed.
I will try adding a get handler and then get back to you

**Comment by jgould22** [comment, marked as answer]
That is not quite how it works.

You must provide some html/javascript for the browser to use to post your file.

You can see a bit of an example in this part of the documentation about posting multiple files 
https://fastapi.tiangolo.com/tutorial/request-files/#multiple-file-uploads-with-additional-metadata
